File: sidecar/app/turn/domain/dtos/turn_dtos.py
"""Turn feature data transfer objects."""

from __future__ import annotations

from dataclasses import dataclass

from sidecar.app.turn.domain.enums.socket_message_type import TurnOutboundSocketEvent


@dataclass
class TurnRequestDto:
    """Normalized inbound turn request payload."""

    session_id: str
    text: str | None
    image_base64: str | None
    audio_base64: str | None


@dataclass
class TurnStreamEventDto:
    """Normalized outbound turn stream event."""

    type: TurnOutboundSocketEvent
    data: object | None = None
    audio: str | None = None
    sample_rate: int | None = None
    sentence_count: int | None = None
    index: int | None = None
    text: str | None = None
    tts_time: float | None = None
    message: str | None = None

File: sidecar/src/app/turn/domain/dtos/turn-dtos.ts
import type { Nullable } from '../../../../shared/types/object';

export type TurnRequestDto = {
  requestId: string;
  text: Nullable<string>;
  imageBase64: Nullable<string>;
  audioBase64: Nullable<string>;
  presetId: string;
};

export type TurnStreamEventDto =
  | { type: "token"; requestId: string; text: string }
  | {
      type: "audio_start";
      requestId: string;
      sampleRate: number;
      sentenceCount: number;
    }
  | { type: "audio_chunk"; requestId: string; audio: string; index?: number }
  | { type: "audio_end"; requestId: string; ttsTime: number }
  | { type: "done"; requestId: string; interrupted?: boolean }
  | { type: "error"; requestId?: string; message: string };
